src/AppBundle/Service/TeamMembershipService.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace AppBundle\Service;
+
+use AppBundle\Event\TeamMembershipEvent;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class TeamMembershipService
+{
+    private $em;
+    private $dispatcher;
+
+    public function __construct(EntityManager $em, EventDispatcherInterface $dispatcher)
+    {
+        $this->em = $em;
+        $this->dispatcher = $dispatcher;
+    }
+
+    public function updateTeamMemberships()
+    {
+        $teamMemberships = $this->em->getRepository('AppBundle:TeamMembership')->findBy(array('isSuspended' => false));
+        foreach ($teamMemberships as $teamMembership) {
+            $endSemester = $teamMembership->getEndSemester();
+            if ($endSemester) {
+                if ($endSemester->getSemesterEndDate() <= $this->em->getRepository('AppBundle:Semester')->findCurrentSemester()->getSemesterStartDate()) {
+                    $teamMembership->setIsSuspended(true);
+                    $this->em->persist($teamMembership);
```

Comment:
Tips: Doctrine overvåker alle entities som hentes ut med `$this->em->getRepository('AppBundle:...')`. Hvis det gjøres en endringen i noen av disse entitiene vil de bli lagret til databasen neste gang `$em->flush()` kalles.

Ved å kalle `$em->persist($obj)` ber man Doctrine om å overvåke `$obj`. Dette må gjøres hvis man f.eks. skal lage nye objekter siden Doctrine ikke allerede vet om disse.

I tilfellet her er det ikke nødvendig å gjøre `$this->em->persist($teamMembership);`, siden Doctrine allerede overvåker denne `$teamMembership` fra `$this->em->getRepository('AppBundle:TeamMembership')->findBy(array('isSuspended' => false));`

https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/change-tracking-policies.html#deferred-implicit